src/typescript/frontend/src/components/Header.tsx

```diff
@@ -55,14 +52,18 @@ const NavItemDivider: React.FC = () => {
 };
 
 type HeaderProps = {
-  allMarketData: ApiMarket[];
   logoHref: string;
+  onDepositWithdrawClick?: MouseEventHandler<HTMLButtonElement>;
+  onWalletButtonClick?: MouseEventHandler<HTMLButtonElement>;
 };
 
-export function Header({ allMarketData, logoHref }: HeaderProps) {
-  const { disconnect } = useWallet();
+export function Header({
+  logoHref,
+  onDepositWithdrawClick,
+  onWalletButtonClick,
```

Comment:
1. i don't agree with handling internally as currently some modal state relies on actions in the header
2. since the component is mostly single use it doesn't make sense to pass in a `WalletConnectButton`. Even if the component were used elsewhere, we wouldn't necessarily want the same behavior to occur anyways.

i think for now best leave it as it is.